code/modules/mob/living/carbon/life.dm

```diff
@@ -235,6 +241,43 @@
 	if(reagents)
 		reagents.metabolize(src)
 

```

Comment:
It's a slightly similar concept with a different execution. This is ten times more lightweight than disease code and does the job just fine.